src/util.c

```diff
@@ -872,6 +872,26 @@ int ld2string(char *buf, size_t len, long double value, ld2string_mode mode) {
     return 0;
 }
 
+/* Parses a version string on the form "major.minor.patch" and returns an
+ * integer on the form 0xMMmmpp. Returns -1 on parse error. */
+int version2num(const char *version) {
```

Comment:
```suggestion
int version2num(const sds version) {
```

Reply:
sds has the length baked in so it is always more efficient to work with sds than c_str.

Reply:
True, but util.c doesn't even include sds.h. All these functions are for `char *` (and sometimes a `size_t len` as a separate argument).